pkg/scheduler/scheduler_test.go

```diff
@@ -994,15 +997,20 @@ func TestFrameworkHandler_IterateOverWaitingPods(t *testing.T) {
 			fakeClient := fake.NewSimpleClientset(objs...)
 			informerFactory := informers.NewSharedInformerFactory(fakeClient, 0)
 			eventBroadcaster := events.NewBroadcaster(&events.EventSinkImpl{Interface: fakeClient.EventsV1()})
+			defer eventBroadcaster.Shutdown()
+
 			eventRecorder := eventBroadcaster.NewRecorder(scheme.Scheme, fakePermit)
 
 			outOfTreeRegistry := frameworkruntime.Registry{
 				fakePermit: newFakePermitPlugin(eventRecorder),
 			}
 
 			_, ctx := ktesting.NewTestContext(t)
-			ctx, cancel := context.WithCancel(ctx)
+			// Theoretically, the timeout should be greater than permitTimeout, but considering
+			// we have no waiting logic in permit plugin, this is fair enough.
+			ctx, cancel := context.WithTimeout(ctx, permitTimeout)
 			defer cancel()
+			tCtx := utiltesting.InitCtx(ctx, t)
```

Comment:
You don't need both. You can remove this:
```
_, ctx := ktesting.NewTestContext(t)
...
ctx, cancel := context.WithTimeout(ctx, permitTimeout)
```

Then below:
```
scheduler, err := New(tCtx, ...
```

This removes the `permitTimeout`. I don't understand why you are adding it to the context of the scheduler.

The test is going to be stuck below in the poll loop until it succeeds or times out there, Giving the scheduler a context with a separate timeout does not change that.

Or was this timeout intended for the polling? Then below use:
```
utiltesting.Eventually(tCtx, ...).Timeout(permitTimeout).Should(...)
```

Reply:
Eventually timeout is all I need.

Reply:
Fixed.